netebpfext/net_ebpf_ext.c

```diff
@@ -870,3 +898,67 @@ net_ebpf_ext_unregister_providers()
         _net_ebpf_sock_ops_providers_registered = false;
     }
 }
+
+ebpf_result_t
+net_ebpf_ext_add_client_context(
+    _Inout_ net_ebpf_extension_wfp_filter_context_t* filter_context,
+    _In_ const struct _net_ebpf_extension_hook_client* hook_client)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    KIRQL old_irql;
+
+    NET_EBPF_EXT_LOG_ENTRY();
+
+    old_irql = ExAcquireSpinLockExclusive(&filter_context->lock);
+
+    // Check if we have reached max capacity.
+    if (filter_context->client_context_count == filter_context->client_context_count_max) {
+        result = EBPF_NO_MEMORY;
```

Comment:
suggest: Should we add an explicit trace here to indicate that this is why the attach failed?

Reply:
Added.